### DISCUSSION TITLE
Select

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#1385] [Feedback]
Use this place to discuss the [select component](https://ons-design-system.netlify.com/components/select) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by trevorsaint** [comment]
We need a select that does not have a set width. The width should be fluid based on its contents.

Similar to languages: https://census-website-respond.netlify.app/end-to-end/languages

**Reply by jrbarnes9** [reply]
Issue #1650 raised

**Comment by jrbarnes9** [comment]
Add a warning panel to 'select' component documentation for the "use with caution" messaging

**Reply by jrbarnes9** [reply]
Issue #1521 raised

**Comment by jrbarnes9** [comment]
Can’t adjust bottom margin class on error panels

**Reply by jrbarnes9** [reply]
Issue #1568 raised

**Comment by jrbarnes9** [comment]
Add inline label variant to select component 

For special cases, such as 'sort by'.

Placing the label above the form control does not work in some situations. Therefore a variant will be required.

Also:
- A normal weight variant for labels would also be useful.

**Reply by jrbarnes9** [reply]
Issue #1649 raised

**Comment by JoelClemence** [comment]
Hello we want to have the select inline with other components like:
![Screenshot 2022-04-12 at 15 08 08](https://user-images.githubusercontent.com/15887464/162981166-bb8f14a6-679e-428d-ad3d-23c0551ad9da.png)

However, I only seem to be able to make the label inline with the select:

![Screenshot 2022-04-12 at 15 09 14](https://user-images.githubusercontent.com/15887464/162981371-3059ebc3-16d1-4889-842e-0ac0f3f55978.png)

**Reply by JoelClemence** [reply]
For clarity, we cannot use the pagination component as we don't know the number of pages up front.

**Reply by JoelClemence** [reply]
We have it working with the style `display: flex-inline`

**Reply by jrbarnes9** [reply]
Hi Joel. Yes, the grid flex utility is currently the only way of doing this.  I've raised a ticket to adapt the pagination component to allow for your use case. #2178

**Comment by jrbarnes9** [comment]
Select.js doesn't handle multiple instances of the component on the same page

**Comment by AmishGudur** [comment]
I just attended the lightning talk about ONS design system and was told to get in touch if any extra components are needed, I am currently working on creating a SelectAll button which selects all items in a dropdown and there is no example of this particular button on the Design system page thus will there be an example of this button be added onto the design system page in the future?

**Reply by MagdalenaLarge** [reply]
Hi @AmishGudur 
Thank you for getting in touch with us. We don't have this functionality in Select component, but we have something similar in the Checkbox: https://service-manual.ons.gov.uk/design-system/components/checkboxes/example-checkboxes-with-select-all-button.
Hopefully this will help with your design, but if not, please let me know and we can discuss your requirement in more details. Thank you.
